packages/util-dynamodb/src/convertToNative.ts

```diff
@@ -41,22 +42,40 @@ export const convertToNative = (data: AttributeValue, options?: unmarshallOption
   throw new Error(`No value defined: ${JSON.stringify(data)}`);
 };
 
-const convertNumber = (numString: string, options?: unmarshallOptions): number | bigint | NumberValue => {
-  if (options?.wrapNumbers) {
+const convertNumber = (
+  numString: string,
+  options?: unmarshallOptions
+): number | bigint | NumberValue | DynamoDBNumber => {
+  if (options?.wrapNumbers && !options?.useDynamoDBNumberWrapper) {
     return { value: numString };
```

Comment:
use `wrapNumbers` directly instead of `useDynamoDBNumberWrapper` and wrap it with DynamoDBNumber renamed to `NumberValue` (see util-dynamodb/models).